test/integration/pipeline/multisig_tx_pipeline_test.cpp

```diff
@@ -157,15 +158,28 @@ TEST_F(MstPipelineTest, OnePeerSendsTest) {
       };
 
   auto &mst_itf = prepareMstItf();
-  mst_itf.sendTx(complete(tx, kUserKeypair), check_mst_pending_tx_status)
+  auto mst_tx = complete(tx, kUserKeypair);
+  mst_itf.sendTx(mst_tx, check_mst_pending_tx_status)
       .sendTx(complete(tx, signatories[0]), check_mst_pending_tx_status)
       .sendTx(complete(tx, signatories[1]),
-              check_enough_signatures_collected_tx_status)
-      .skipProposal()
-      .skipVerifiedProposal()
-      .checkBlock([](auto &proposal) {
-        ASSERT_EQ(proposal->transactions().size(), 1);
-      });
+              check_enough_signatures_collected_tx_status);
+
+  // wait until multisig transaction is appeared
+  while (true) {
+    auto required_break = false;
+    mst_itf.checkProposal([&mst_tx, &required_break](auto &proposal) {
+      ASSERT_EQ(proposal->transactions().size(), 1);
+      auto &proposal_tx = *boost::begin(proposal->transactions());
+      if (mst_tx.hash() == proposal_tx.hash()) {
+        ASSERT_EQ(boost::size(proposal_tx.signatures()), 3);
```

Comment:
How did we get 3 signatures here? I see only that this transaction has been signed twice